fxa/plugins/requests.py

```diff
@@ -3,17 +3,30 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 from __future__ import absolute_import
 
-from binascii import hexlify
-from hashlib import sha256
+import json
 import os
 
+from binascii import hexlify
+from hashlib import sha256
 from requests.auth import AuthBase
+from six import text_type
 from six.moves.urllib.parse import urlparse
 
+from fxa.cache import MemoryCache
 from fxa import core
 from fxa import oauth
 
 DEFAULT_CLIENT_ID = "5882386c6d801776"  # Firefox dev Client ID
+DEFAULT_CACHE_EXPIRY = 3600
+
+
+def get_cache_key(*args):
+    cache_key = sha256()
+    for key in args:
+        if key:
+            cache_key.update(text_type(key).encode('utf-8'))
+        cache_key.update(b'\n')
```

Comment:
those lines probably deserve some testing love :)

...like «*will the tests fail if I remove them ?*»